packages/client/hmi-client/src/utils/optimize.ts

```diff
@@ -138,63 +139,232 @@ export function createOptimizeChart(
-						},
-						scale: { padding: 10 }
-					},
-					color: {
-						field: 'phase',
-						type: 'nominal',
-						scale: { range: ['#a3a3a3', '#1B8073'] },
-						legend: { title: null },
-						sort: ['Before optimization', 'After optimization']
-					}
-				}
-			},
-			...createInterventionChartMarkers(data)
-		]
-	};
-}
-
-function createInterventionChartMarkers(
-	data: { name: string; value: number; time: number; phase: string }[]
-) {
+function createInterventionChartMarkers(data: { name: string; value: number; time: number }[]) {
 	return data.map((ele) => ({
+		data: [{}], // Dummy data to ensure the layer is rendered
 		mark: { type: 'rule', strokeDash: [4, 4], color: 'black' },
 		encoding: {
 			x: { datum: ele.time }
 		}
 	}));
 }
+
+export const createOptimizeForecastChart = (
+	preSampleRunData: Record<string, any>[],
+	preStatisticData: Record<string, any>[],
+	postSampleRunData: Record<string, any>[],
+	postStatisticData: Record<string, any>[],
+	interventionsData: { name: string; value: number; time: number }[],
+	options: any
+) => {
+	const axisColor = '#EEE';
+	const labelColor = '#667085';
+	const labelFontWeight = 'normal'; // Adjust font weight here
+	const titleObj = options.title
+		? {
+				text: options.title,
+				anchor: 'start',
+				subtitle: ' ',
+				subtitlePadding: 4
+			}
+		: null;
+
+	const xaxis = {
+		domainColor: axisColor,
+		tickColor: { value: axisColor },
+		labelColor: { value: labelColor },
+		labelFontWeight,
+		title: options.xAxisTitle,
+		gridColor: '#EEE',
+		gridOpacity: 1.0
+	};
+	const yaxis = structuredClone(xaxis);
+	yaxis.title = options.yAxisTitle;
+
+	const spec: any = {
+		$schema: VEGALITE_SCHEMA,
+		title: titleObj,
+		description: '',
+		width: options.width,
+		height: options.height,
+
+		// layers
+		layer: [],
+
+		// Make layers independent
+		resolve: {
+			legend: { color: 'independent' },
+			scale: { color: 'independent' }
+		}
+	};
+
+	// Build pre sample layer
+	if (preSampleRunData && preSampleRunData.length > 0) {
+		const sampleVariables = options.variables?.map((d) => d);
+
+		spec.layer.push({
+			mark: { type: 'line' },
+			data: { values: preSampleRunData },
+			transform: [
+				{
+					fold: sampleVariables,
+					as: ['sample_variable', 'sample_value']
+				}
+			],
+			encoding: {
+				x: { field: options.timeField, type: 'quantitative', axis: xaxis },
+				y: { field: 'sample_value', type: 'quantitative', axis: yaxis },
+				color: {
+					field: 'sample_variable',
+					type: 'nominal',
+					scale: {
+						domain: sampleVariables,
+						range: ['#AAB3C6']
+					},
+					legend: false // No legend for sampling-layer, too noisy
+				},
+				detail: { field: options.groupField, type: 'nominal' },
+				strokeWidth: { value: 1 },
+				opacity: { value: 0.1 }
+			}
+		});
+	}
+
+	// Build post sample layer
+	if (postSampleRunData && postSampleRunData.length > 0) {
+		const sampleVariables = options.variables?.map((d) => d);
+
+		spec.layer.push({
```

Comment:
If your pre and post layers have a lot of similarities maybe we can build couple of helper functions to avoid repeating code.

Reply:
Ive created 2 function to create sample and statistic layers